ground/src/main/java/com/google/android/ground/persistence/sync/LocalMutationSyncWorker.kt

```diff
@@ -123,22 +145,18 @@ constructor(
       .map { (_, _, newResponse): TaskDataDelta -> newResponse.toString() }
       .forEach { remotePath: String -> photoSyncWorkManager.enqueueSyncWorker(remotePath) }
 
-  private fun groupByUserId(pendingMutations: List<Mutation>): Map<String, List<Mutation>> =
-    pendingMutations.groupBy { it.userId }
-
-  private fun incrementRetryCounts(mutations: List<Mutation>, error: Throwable): List<Mutation> =
-    mutations.map { m: Mutation -> incrementRetryCount(m, error) }
-
-  private fun incrementRetryCount(mutation: Mutation, error: Throwable): Mutation =
-    when (mutation) {
-      is LocationOfInterestMutation ->
-        mutation.copy(retryCount = mutation.retryCount + 1, lastError = error.toString())
-      is SubmissionMutation ->
-        mutation.copy(retryCount = mutation.retryCount + 1, lastError = error.toString())
+  private suspend fun getUser(userId: String): User? {
+    val user = localUserStore.getUserOrNull(userId)
+    if (user == null) {
+      Timber.e("User account removed before mutation processed")
     }
+    return user
+  }
 
   companion object {
-    private const val LOCATION_OF_INTEREST_ID_PARAM_KEY = "locationOfInterestId"
+    private const val MAX_RETRY_COUNT = 3
```

Comment:
3 seems like too few if the user has a spotty connection. Can we set this to a higher number, say, "10", and ensure that the worker is using exponetional backoff for retries?

Reply:
Done